### ISSUE TITLE
Getting Empty URL 

### ISSUE THREAD

**Issue by RupaSiddana** [soachishti/moss.py#34]
Report URL: 
Traceback (most recent call last):
  File "/home/ib-developer/Desktop/Roopa/Data_Analytics_utils/MOSS/moss.py-1.0.8/moss_usage.py", line 20, in <module>
    m.saveWebPage(url, "MOSS/moss.py-1.0.8/submission/report.html")
  File "/home/ib-developer/Desktop/Roopa/Data_Analytics_utils/MOSS/moss.py-1.0.8/mosspy/moss.py", line 144, in saveWebPage
    raise Exception("Empty url supplied")
Exception: Empty url supplied

I used this code
```
import mosspy

userid = 987654321

m = mosspy.Moss(userid, "python")

m.addBaseFile("MOSS/moss.py-1.0.8/submission/a01.py")
m.addBaseFile("MOSS/moss.py-1.0.8/submission/test_student.py")

# Submission Files
m.addFile("MOSS/moss.py-1.0.8/submission/a01-sample.py")
m.addFilesByWildcard("MOSS/moss.py-1.0.8/submission/a01-*.py")


url = m.send()

print ("Report URL: " + url)

# Save report file
m.saveWebPage(url, "MOSS/moss.py-1.0.8/submission/report.html")

mosspy.download_report(url, "submission/report/", connections=8, log_level=10) # logging.DEBUG (20 to disable)
```

**Comment by soachishti** [comment]
Hi @RupaSiddana, please ask your question here. 

Also, have you received your _userid_ from moss?

FYI, the step for registering is available at http://theory.stanford.edu/~aiken/moss/ under the heading "Registering for Moss".

**Comment by RupaSiddana** [comment]
No , i did not received

**Comment by RupaSiddana** [comment]
What i do after i receive the user_id

**Comment by soachishti** [comment]
When you receive that, clone this repo locally and run moss_usage.py with your userid.

**Comment by RupaSiddana** [comment]
Yesterday i  kept mail at 11:28 
It has been 8 hours completed 
still i did not get any mail from moss
what can i do

**Comment by soachishti** [comment]
I just give it a try and it seems to respond. Kindly verify that the email body is in the text without any formatting eg. bold, italic, and etc.

**Comment by RupaSiddana** [comment]
registeruser
mail roopa.siddana@gmail.com

**Comment by RupaSiddana** [comment]
Yeah i got email, i run the moss_usage.py with my user id
But i did not get report html 
Please Explain what is use of add base file and add file

**Comment by soachishti** [comment]
You will receive the report URL in line 20 of `moss_usage.py`.  https://github.com/soachishti/moss.py/blob/b82024558ebfe7993ca59740d8df8c2606c04454/moss_usage.py#L20

For understanding concepts refer to the following links;

1. Comments in this script; http://moss.stanford.edu/general/scripts/mossnet
2. Moss paper; http://theory.stanford.edu/~aiken/publications/papers/sigmod03.pdf

**Comment by RupaSiddana** [comment]
still issue resolved
``
m = mosspy.Moss(userid, "python")

m.addBaseFile("/home/ib-developer/Desktop/Roopa/Data_Analytics_utils/MOSS/moss.py-1.0.8/submissions/base_code.py", "base_code")


#Submission Files
csv_file_folder_path = "/home/ib-developer/Desktop/Roopa/Data_Analytics_utils/MOSS/moss.py-1.0.8/submissions"
csv_files_list = [
    csv_file for csv_file in os.listdir(csv_file_folder_path)
    if csv_file.endswith('.py') and csv_file.startswith('user')
]
csv_files_list = sorted(csv_files_list)
for csv_file in csv_files_list:
    file_path = csv_file_folder_path+"/"+csv_file
    display_name = csv_file.replace('.csv',"")
    m.addFile(file_path, display_name)


url = m.send()

print("Report URL: " + url)

# Save report file
m.saveWebPage(url, "/home/ib-developer/Desktop/Roopa/Data_Analytics_utils/MOSS/moss.py-1.0.8/submissions/report.html")

mosspy.download_report(url, "submissions/report/", connections=8, log_level=10) # logging.DEBUG (20 to disable)``


iam getting result  like

File 1 | File 2 | Lines Matched
-- | -- | --
user4_submission.py (90%) | user5_submission.py (90%) | 8
user3_submission.py (90%) | user5_submission.py (90%) | 8
user3_submission.py (90%) | user4_submission.py (90%) | 8
user2_submission.py (90%) | user5_submission.py (90%) | 8
user2_submission.py (90%) | user4_submission.py (90%) | 8
user2_submission.py (90%) | user3_submission.py (90%) | 8
user1_submission.py (90%) | user5_submission.py (90%) | 8
user1_submission.py (90%) | user4_submission.py (90%) | 8
user1_submission.py (90%) | user3_submission.py (90%) | 8
user1_submission.py (90%) | user2_submission.py (90%) | 8

**Here what i want is File 1 should be always my base file**

**Comment by soachishti** [comment]
What I can see here is you are setting `base_code.py` as the base file and all `user*.py` as submission file.

In the report, the result shows that `user4_submission.py` has a 90% match with `user5_submission.py` and so on. This all looks fine to me.

Are you looking for an approach to add `user1_submission.py` as a base file too?

**Comment by soachishti** [comment]
@RupaSiddana Closing this issue. Feel free to open if you are still facing the problem.

**Comment by ayanw** [comment]
Hello @soachishti, 

I have been using Moss for some time but for one file I keep getting an empty url. 
I cannot find the solution in this thread but I am also getting 'Exception: Empty url supplied' if I use Moss from the script. 
And if I use it from the app, I get this pop-up: 
![image](https://user-images.githubusercontent.com/50455002/150297848-58fb25ee-878f-43d7-a4ff-2fc41427050a.png)

**Comment by soachishti** [comment]
@ayanw Thanks for reaching out. It seem like is working fine for you in most cases. Normally this happens due to network issue, bad request or missing user id.

If you are seeing this case consistently for a giving file. Can you share more details (with example file) in a separate issue.

**Comment by bdr** [comment]
I'm having a problem where "url = m.send(lambda file_path, display_name: print('', end='', flush=True))" returns an empty URL.  I discovered that when I remove a file from the collection of files sent to MOSS, all is well.  I've attached the file
[cell-a8c22df1236cd837@89886910094@1623940299680.txt](https://github.com/soachishti/moss.py/files/8097478/cell-a8c22df1236cd837%4089886910094%401623940299680.txt)
p.s.  this file actually ends with ".source" extension, but github refused to attach a file with that suffix!

If I comment out the following line in this file:
`df4 = pd.DataFrame(pca.components_[0:2], columns = feature_names, index=['PC1','PC2'])` 
then everything works correctly.   I'm not sure why the inclusion of this file causes mosspy to fail.  All works well when submitting directly to moss:
`./moss -l python -d /Users/bdr/antiplag/kct-zips/543s21ASSN1TST2000/*.source
`
any advice?
